### ISSUE TITLE
applyBackground issue

### ISSUE THREAD

**Issue by zsoooc** [openlayers/ol-mapbox-style#1040]
I would like to change the style of the backround layer of my map using applyBackground function. However, it doesn't work.
The default MapTiler style is "basic-v2", and I want to change the style to "topo-v2" from the browser (Firefox) console. 
If I run this: `olms.applyBackground(map,'https://api.maptiler.com/maps/topo-v2/style.json?key=xxx') 
`
The map disappears and the background color changes.

![image](https://github.com/openlayers/ol-mapbox-style/assets/19270270/44567013-1506-4fd4-9b07-cb4e985ff727)

 
This is my code:
```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="initial-scale=1,maximum-scale=1,user-scalable=no" />
    <title>Display a map</title>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/ol@v8.1.0/ol.css">
    <script src="https://cdn.jsdelivr.net/npm/ol@v8.1.0/dist/ol.js"></script>
    <script src="https://unpkg.com/ol-mapbox-style@12.0.0/dist/olms.js"></script>
    <style>
      #map {position: absolute; top: 0; right: 0; bottom: 0; left: 0;}
    </style>
  </head>
  <body>
    <div id="map">
      <a href="https://www.maptiler.com" style="position:absolute;left:10px;bottom:10px;z-index:999;"></a>
    </div>
    <script>
      const key = 'B36QTUMdhsxLLI5w8xcL';

let basicMap = new olms.MapboxVectorLayer({
      styleUrl: 'https://api.maptiler.com/maps/basic-v2/style.json?key=xxx'
    });
let topoMap = new olms.MapboxVectorLayer({
      styleUrl: 'https://api.maptiler.com/maps/topo-v2/style.json?key=xxx'
    });

const map = new ol.Map({
        layers: [basicMap],
        target: 'map',
        view: new ol.View({
          constrainResolution: true,
          center: ol.proj.fromLonLat([16.62662018, 46.2125578]),
          zoom: 14,
          maxZoom:19
        })
      });

    </script>
  </body>
</html>
```

**Comment by ahocevar** [comment]
Thanks for reporting this. Will be fixed with #1041 In the meantime, call 
```js
const layer = map.getLayers().item(0);
applyBackground(layer,'https://api.maptiler.com/maps/topo-v2/style.json?key=xxx');
```

**Comment by ahocevar** [comment]
Note that satellite is a raster layer, not a vector layer. The background is in the image that you get from the server - it cannot be changed.
